lib/Command/Prompt.php

```diff
@@ -43,16 +44,59 @@ protected function configure() {
 				InputArgument::REQUIRED,
 				'The prompt'
 			)
-			->addOption('no-context', null, InputOption::VALUE_NONE, 'Do not use context');
+			->addOption(
+				'no-context',
+				null,
+				InputOption::VALUE_NONE,
+				'Do not use context'
+			)
+			->addOption(
+				'context-sources',
+				null,
+				InputOption::VALUE_REQUIRED,
+				'Context sources to use (as a comma-separated list without brackets)',
+			)
+			->addOption(
+				'context-providers',
+				null,
+				InputOption::VALUE_REQUIRED,
+				'Context providers to use (as a comma-separated list without brackets)',
+			);
 	}
 
 	protected function execute(InputInterface $input, OutputInterface $output) {
 		$userId = $input->getArgument('uid');
 		$prompt = $input->getArgument('prompt');
 		$noContext = $input->getOption('no-context');
+		$contextSources = $input->getOption('context-sources');
+		$contextProviders = $input->getOption('context-providers');
+
+		if ($noContext && (!empty($contextSources) || !empty($contextProviders))) {
+			throw new \InvalidArgumentException('Cannot use --no-context with --context-sources or --context-provider');
+		}
+
+		if (!empty($contextSources) && !empty($contextProviders)) {
+			throw new \InvalidArgumentException('Cannot use --context-sources with --context-provider');
+		}
 
 		if ($noContext) {
 			$task = new Task(FreePromptTaskType::class, $prompt, 'context_chat', $userId);
+		} elseif (!empty($contextSources)) {
+			$contextSources = preg_replace('/\s*,+\s*/', ',', $contextSources);
+			$contextSourcesArray = array_filter(explode(',', $contextSources), fn ($source) => !empty($source));
+			$task = new Task(ScopedContextChatTaskType::class, json_encode([
+				'scopeType' => 'source',
+				'scopeList' => $contextSourcesArray,
+				'prompt' => $prompt,
+			]), 'context_chat', $userId);
+		} elseif (!empty($contextProviders)) {
+			$contextProviders = preg_replace('/\s*,+\s*/', ',', $contextProviders);
+			$contextProvidersArray = array_filter(explode(',', $contextProviders), fn ($source) => !empty($source));
+			$task = new Task(ScopedContextChatTaskType::class, json_encode([
+				'scopeType' => 'provider',
```

Comment:
Can we still have these as constants? Would be nice

Reply:
having them as constants wouldn't change much since the check in the text-processing provider still has to check for them manually. It was easier and made sense with enum but not sure now.  
Shall we use a class with readonly variables here?

Reply:
Maybe a class with public constants? Setting hardcoded values as constants is always nice, IMO, since it gives you static analysis visibility, IDE auto-complete and other goodies :)